spring-integration-http/src/main/java/org/springframework/integration/http/dsl/HttpMessageHandlerSpec.java

```diff
@@ -75,8 +75,7 @@ public HttpMessageHandlerSpec requestFactory(ClientHttpRequestFactory requestFac
 	 * @return the spec
 	 */
 	public HttpMessageHandlerSpec errorHandler(ResponseErrorHandler errorHandler) {
-		Assert.isTrue(this.isClientSet(),
-				"the 'errorHandler' must be specified on the provided 'restTemplate'");
+		Assert.isTrue(!isClientSet(), "the 'errorHandler' must be specified on the provided 'restTemplate'");
```

Comment:
Good. Thank you!

Would you mind to modify `HttpDslTests` to cover this change?

Also consider, please, fix `requestFactory()` condition above to use this `isClientSet()` as well.

And add you name to the `@author` list of this class and test one, if you really are going to modify that as well.